code/main.py

```diff
@@ -63,31 +54,52 @@ def main():
     service_principal_id=azure_credentials.get("clientId", "")
     service_principal_password=azure_credentials.get("clientSecret", "")
     subscriptionId=azure_credentials.get("subscriptionId", "")
-    command = ("az login --service-principal --username {APP_ID} --password \"{PASSWORD}\" --tenant {TENANT_ID}").format(
-            APP_ID=service_principal_id, PASSWORD=service_principal_password, TENANT_ID=tenant_id)
-    try:
-        login_result = subprocess.check_output(command, shell=True)
-        print(login_result)
-    except Exception as ex:
-        print(ex)
-        return;
-
+    
     success = False
+    jsonobject = None
     try:
```

Comment:
we can move this part to a utility function

Reply:
moved to utils